src/Mono.Android/Android.App/Activity.cs

```diff
@@ -22,6 +28,29 @@ public void RunOnUiThread (Action action)
 		{
 			RunOnUiThread (new Java.Lang.Thread.RunnableImplementor (action));
 		}
+
+		protected T SetContentView <T> () where T: LayoutBindingBase, new()
+		{
+			T items = Activator.CreateInstance (
+				typeof (T),
+				BindingFlags.CreateInstance | BindingFlags.DeclaredOnly | BindingFlags.NonPublic | BindingFlags.Public,
+				null,
+				new Object[] {this as ILayoutBindingClient},
+				null
+			) as T;
```

Comment:
This is likewise weird. [`Activator.CreateInstance()`](https://msdn.microsoft.com/en-us/library/2ca9yys8(v=vs.110).aspx) *must* create a value of the `type` parameter; if it doesn't, that Just Weird™.

Why not just cast the return value?

Reply:
Just Weird™ things happen all the time, as we know, and I see no reason to leave a way for them to happen if we can prevent it. In this case, I think it's better a `null` is returned than an exception is thrown should the returned type be, for some reason, not  correct. But, if you insist, I can just make it a cast.